pkg/timelock/timelock.go

```diff
@@ -35,7 +35,7 @@ type Worker struct {
 	ethClient       *ethclient.Client
```

Comment:
I would recommend that you use an interface defintition as the dependency instead of the *ethclient.Client.  there aren't any unit tests to validate the refactor.  The first thing we would want to do is abstract away `FilterQUery` and `SubscribeFilterQuery` so that the worker could have a few tests on `Listen`